code/modules/ruins/lavalandruin_code/elephantgraveyard.dm

```diff
@@ -0,0 +1,234 @@
+	environment_type = "wasteland"
+	baseturfs = /turf/open/floor/plating/asteroid/basalt/wasteland
+	digResult = /obj/item/stack/ore/glass/basalt
+	initial_gas_mix = OPENTURF_LOW_PRESSURE
+	slowdown = 0.5
+	floor_variance = 30
+
+/turf/closed/mineral/strong/wasteland
+	name = "Ancient dry rock"
+	environment_type = "wasteland"
+	turf_type = /turf/open/floor/plating/asteroid/basalt/wasteland
+	baseturfs = /turf/open/floor/plating/asteroid/basalt/wasteland
+	initial_gas_mix = LAVALAND_DEFAULT_ATMOS
+	defer_change = 1
+	smooth_icon = 'icons/turf/walls/rock_wall.dmi'
+
+/turf/open/floor/plating/asteroid/basalt/wasteland/Initialize()
+	.=..()
+	if(prob(floor_variance))
+		icon_state = "[environment_type][rand(0,6)]"
+
+//***Oil well puddles.
+/obj/structure/sink/oil_well	//Not going to enjoy bathing in this...
+	name = "oil well"
+	desc = "A bubbling pool of oil. Would probably be valuable, had bluespace technology not destroyed the need for fossil fuels 200 years ago."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "puddle-oil"
+	dispensedreagent = /datum/reagent/fuel/oil
+
+/obj/structure/sink/oil_well/attack_hand(mob/M)
+	icon_state = "puddle-oil-splash"
+	. = ..()
+	icon_state = "puddle-oil"
+
+/obj/structure/sink/oil_well/attackby(obj/item/O, mob/user, params)
+	icon_state = "puddle-oil-splash"
+	. = ..()
+	icon_state = "puddle-oil"
+
+/obj/structure/sink/oil_well/deconstruct(disassembled = TRUE)
+	qdel(src)
+
+//***Grave mounds.
+/obj/structure/closet/crate/grave
+	name = "burial mound"
+	desc = "An marked patch of soil, showing signs of a burial long ago. You wouldn't disturb a grave... right?"
+	icon = 'icons/obj/crates.dmi'
+	icon_state = "grave"
+	dense_when_open = TRUE
+	material_drop = /obj/item/stack/ore/glass/basalt
+	material_drop_amount = 5
+	anchorable = FALSE
+	anchored = TRUE
+	locked = TRUE
+	breakout_time = 900
+	cutting_tool = /obj/item/shovel
+	var/lead_tomb = FALSE
+	var/first_open = FALSE
+
+/obj/structure/closet/crate/grave/PopulateContents()  //GRAVEROBBING IS NOW A FEATURE
+	..()
+	switch(rand(1,8))
+		if(1)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/coin/gold(src)
+			new /obj/item/storage/wallet(src)
+		if(2)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/clothing/glasses/meson(src)
+		if(3)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/coin/silver(src)
+			new /obj/item/shovel/spade(src)
+		if(4)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/storage/book/bible/booze(src)
+		if(5)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/clothing/neck/stethoscope(src)
+			new	/obj/item/scalpel(src)
+			new /obj/item/hemostat
+		if(6)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/storage/box/beakers(src)
+			new /obj/item/clothing/glasses/science(src)
+		if(7)
+			new /obj/effect/mob_spawn/human/skeleton
+			new /obj/item/clothing/glasses/sunglasses(src)
+			new /obj/item/clothing/mask/cigarette/rollie(src)
```

Comment:
If all the cases have a skeleton, shouldn't you spawn it before the switch case to avoid duplicate code?

Reply:
That's what I tried at first, but it was doing some screwy things with spawning outside of the grave (Yes I know I left out (src) in their spawns). I'm testing having it  just in the map instead since this is the only instance where it'll appear anyway, since it's proving to be a bit picky.